Bigtable/src/BigtableClient.php

```diff
@@ -0,0 +1,191 @@
+     *           Ex: `json_decode(file_get_contents($path), true)`.
+     *     @type string $keyFilePath The full path to your service account
+     *           credentials .json file retrieved from the Google Developers
+     *           Console.
+     *     @type float $requestTimeout Seconds to wait before timing out the
+     *           request. **Defaults to** `0` with REST and `60` with gRPC.
+     *     @type int $retries Number of retries for a failed request.
+     *           **Defaults to** `3`.
+     *     @type array $scopes Scopes to be used for the request.
+     * }
+     * @throws GoogleException If the gRPC extension is not enabled.
+     */
+    public function __construct(array $config = [])
+    {
+        $this->requireGrpc();
+        $config += [
+            'scopes' => [
+                self::FULL_CONTROL_SCOPE,
+                self::ADMIN_SCOPE
+            ],
+            'projectIdRequired' => true
+        ];
+        $this->connection = new Grpc($this->configureAuthentication($config));
+        $this->setLroProperties(new LongRunningConnection($this->connection), [
+            [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.UpdateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ], [
+                'typeUrl' => 'type.googleapis.com/google.bigtable.admin.instance.v2.CreateInstanceMetadata',
+                'callable' => function ($instance) {
+                    $name = InstanceAdminClient::parseName($instance['name'])['instance'];
+                    return $this->instance($name, $instance);
+                }
+            ]
+        ]);
+    }
+
+    /**
+     * Create a new instance.
+     *
+     * Example:
+     * ```
+     * $operation = $bigtable->createInstance('my-instance');
+     * ```
+     *
+     * @codingStandardsIgnoreStart
+     * @see https://cloud.google.com/bigtable/docs/reference/admin/rpc/google.bigtable.admin.v2#CreateInstanceRequest CreateInstanceRequest
+     *
+     * @param string $instanceId The instance ID.
+     * @param array $options [optional] {
+     *     Configuration options
+     *
+     *     @type string $displayName **Defaults to** the value of $name.
+     *     @type int $instanceType **Defaults to** `0`.
+     *     @type array $labels For more information, see
+     *           [Using labels to organize Google Cloud Platform resources](https://cloudplatform.googleblog.com/2015/10/using-labels-to-organize-Google-Cloud-Platform-resources.html).
+     *     @type array $clusters []{
+     *           array {
+     *                 string $clusterId
+     *                 string $locationId
+     *                 int $nodeCount **Defaults to** `1`.
+     *                 int $storageType **Defaults to** `0`.
```

Comment:
storageType should be an enum.

Reply:
PHP doesn't have an enum type. 

In Cloud Spanner, we handle simple cases like this, where there are few possible values by adding constants to the Instance class. So `Google\Cloud\Bigtable\Instance` would have some new constants, `STORAGE_TYPE_SSD`, `STORAGE_TYPE_HDD`. The values of each of these would point to the equivalent value in the proto.

i.e.
```php
const STORAGE_TYPE_SSD = \Google\Cloud\Bigtable\Admin\V2\StorageType::SSD;
```

Then the argument description would list possible values and the default using the constant names:

```
int $storageType The storage media type for persisting Bigtable data. Possible values include `Instance::STORAGE_TYPE_SSD` and `Instance::STORAGE_TYPE_HDD`. **Defaults to** `Instance::STORAGE_TYPE_UNSPECIFIED`.
```